src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -612,7 +612,8 @@ private BoundExpression UnwrapCollectionExpressionIfNullable(BoundExpression col
                     return BoundCall.Synthesized(
                         syntax: exprSyntax,
                         receiverOpt: collectionExpr,
-                        method: nullableValueGetter);
+                        method: nullableValueGetter,
+                        binder: this);
```

Comment:
>binder: this [](start = 24, length = 12)

Is the problem that we didn't pass a binder when it was needed? Can we add an assert somewhere that we got null binder when non-null is expected to be passed?

Reply:
> Is the problem that we didn't pass a binder when it was needed?

Yes, it's needed for `DeriveArguments`.

> Can we add an assert somewhere that we got null binder when non-null is expected to be passed?

It's actually an assertion in `DeriveArguments` that caught this, we just didn't have a test case that verified the IOperation tree for this code path. I'm going to need to incorporate this change into the branch changes. We still have the bug in master, but the assertions I added when nullable enabling are blocking my ability to run benchmarks.

---
In reply to: [530043147](https://github.com/dotnet/roslyn/pull/49534#discussion_r530043147) [](ancestors = 530043147)

Reply:
This iteration is no longer needed.

---
In reply to: [533822002](https://github.com/dotnet/roslyn/pull/49534#discussion_r533822002) [](ancestors = 533822002,530043147)